trin-core/src/portalnet/overlay_service.rs

```diff
@@ -1473,52 +1487,55 @@ where
         Ok(closest_nodes)
     }
 
-    fn init_find_nodes_query_with_initial_enrs(&mut self, target: &NodeId, enrs: Vec<Enr>) {
-        let mut query_info = QueryInfo {
+    fn init_find_nodes_query_with_initial_enrs_and_config(
+        &mut self,
+        target: &NodeId,
+        enrs: Vec<Enr>,
+        query_config: QueryConfig,
+    ) {
+        let query_info = QueryInfo {
             query_type: QueryType::FindNode(*target),
-            untrusted_enrs: Default::default(),
+            untrusted_enrs: SmallVec::from_vec(enrs),
             callback: None,
             distances_to_request: self.findnodes_query_distances_per_peer,
         };
 
-        let mut known_closest_peers = Vec::new();
-        for enr in enrs {
-            // Add the known ENRs to the untrusted list
-            query_info.untrusted_enrs.push(enr.clone());
-            // Add the key to the list for the query
-            let node_id = enr.node_id();
-            let key = Key::from(node_id);
-            known_closest_peers.push(key);
-        }
+        let known_closest_peers: Vec<Key<NodeId>> = query_info
+            .untrusted_enrs
+            .iter()
+            .map(|enr| Key::from(enr.node_id()))
+            .collect();
 
         if known_closest_peers.is_empty() {
             warn!(
                 "FindNodes query initiated but no closest peers in routing table. Aborting query."
             );
         } else {
-            let query_config = QueryConfig {
-                parallelism: self.query_parallelism,
-                num_results: self.query_num_results,
-                peer_timeout: self.query_peer_timeout,
-            };
             let find_nodes_query =
                 FindNodeQuery::with_config(query_config, query_info.key(), known_closest_peers);
-            self.queries.add_query(query_info, find_nodes_query);
+            self.query_pool.add_query(query_info, find_nodes_query);
         }
     }
 
+    fn init_find_nodes_query_with_initial_enrs(&mut self, target: &NodeId, enrs: Vec<Enr>) {
+        let query_config = QueryConfig {
+            parallelism: self.query_parallelism,
+            num_results: self.query_num_results,
+            peer_timeout: self.query_peer_timeout,
+        };
+
+        self.init_find_nodes_query_with_initial_enrs_and_config(target, enrs, query_config);
```

Comment:
It seems like we only added this latter method to use in a test, which I do not think is a good reason to add a function. I would be in favor of reverting to a single method here.

Reply:
@jacobkaufmann Good point, I moved this function to the test module (at the cost of a bit of duplication) for now. In a subsequent PR I'm going to refactor the query config passing/defaults which will allow for having one simple method for initialization find nodes queries rather 3 separate methods for that build on each other for 3 separate cases.